tests/test_placeholder_validator.py

```diff
@@ -189,3 +190,62 @@ def test_validation_answer_and_transform_unit_match(
         arguments=arguments, transform_type=transform_type
     )
     assert validator.errors == expected_error
+
+
+def test_validation_answer_and_transform_unit_two_source_answers_mismatch():
+    filename = "schemas/invalid/test_invalid_calculated_summary_units_mismatch.json"
+    schema_file = _open_and_load_schema_file(filename)
+    validator = PlaceholderValidator(schema_file)
+    validator.validate()
+    assert validator.errors == [
+        {
+            "message": error_messages.ANSWER_UNIT_AND_TRANSFORM_UNIT_MISMATCH.format(
+                answer_unit="length-centimeter", transform_unit="length-meter"
+            ),
+            "identifier": "unit-total-playback",
+        },
+        {
+            "message": error_messages.ANSWER_UNIT_AND_TRANSFORM_UNIT_MISMATCH.format(
+                answer_unit="length-centimeter", transform_unit="length-meter"
+            ),
+            "identifier": "unit-total-playback",
+        },
+    ]
+
+
+def test_validation_answer_and_transform_unit_single_source_answer_mismatch():
+    filename = "schemas/invalid/test_invalid_calculated_summary_units_mismatch.json"
+    schema_file = _open_and_load_schema_file(filename)
+    schema = QuestionnaireSchema(schema_file)
+    answer = schema.get_answer("second-number-answer-unit-total")
```

Comment:
you could probably parametrize these two tests with an `answer_id` parameter as its the only thing that changes

Reply:
It is slightly more complicated, I had to add answer_id and mismatched anwer_id as parameters, since we add the change from this comment: https://github.com/ONSdigital/eq-questionnaire-validator/pull/204#discussion_r1234057634

Reply:
We might have to keep it as it was, cause if the source is answer then error message has duplicates:
`{'message': "The answer unit and transform unit mismatch, 'mile' not 'meter'", 'identifier': 'average-distance', 'answer_id': 'average-distance'}`

Reply:
Check https://github.com/ONSdigital/eq-questionnaire-validator/pull/204#discussion_r1234859166

Reply:
We are already in a loop for a single answer id (when the error happens), so maybe we should keep it as it is to not overcomplicate it and keep it similar to other errors? We would have to build a list to pass into the final message, wouldn't this be over-engineered? I think it's more like errors in validator where you add one for each occurrence but I am happy to change it.

Reply:
Also for calculated it would have to be a different error message instead of answer_id field it would be answer_ids.

Reply:
The question is: when you have a mismatch, do you want an error message for each mismatch?

Reply:
hmm, I see the point, I just don't like that we've got two absolutely identical error messages
```
[{'identifier': 'unit-total-playback',
  'message': "The answer unit and transform unit mismatch, 'length-centimeter' "
             "not 'length-meter'"},
 {'identifier': 'unit-total-playback',
  'message': "The answer unit and transform unit mismatch, 'length-centimeter' "
             "not 'length-meter'"}]
```
because each answer caused another error, but they're in the same block and only that is included in the message

If you omit `answer_id` can the error show just once? perhaps intead of iterating over answers and adding the error, you could do a list comprehension that gets the units, and then if any unit doesn't match the transform one, you just add a single error

Reply:
I think an error for each mismatch is helpful if we know why each mismatch is different or what each one is, otherwise for duplicate errors I think its better to only show it once

Reply:
Discussed in ad-hoc session.